### PR TITLE
feat: SQLite support via JDBC

### PR DESCRIPTION
**Pull request by laysakura**
This is a base (huge) pull request to fully support SQLite.
See separate pull requests for code review.

- #763 
- #764 
- #765 
- #769 
- #819

### CODE REVIEW

**Comment by laysakura** [comment]
The SQLite support finally got merge ready!

File: core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitAdminIntegrationTestWithJdbcDatabase.java [hunk 1]
```diff
@@ -1,13 +1,49 @@
...
   protected Properties getProps(String testName) {
     return JdbcEnv.getProperties(testName);
   }
+
+  // Since SQLite doesn't have persistent namespaces, some behaviors around the namespace are
+  // different from the other adapters. So disable several tests that check such behaviors.
+
+  private boolean isSqlite() {
+    String jdbcUrl = System.getProperty(PROP_JDBC_URL, DEFAULT_JDBC_URL);
+    return jdbcUrl.startsWith("jdbc:sqlite:");
```
Comment (brfrn169) [inline core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitAdminIntegrationTestWithJdbcDatabase.java:25]: Sorry, I missed this in the previous PR for some reason. Can you please add a method doing this in the `JdbcEnv` class, and call the new method here?

**Review by brfrn169 — COMMENTED** [review]
Left a comment. Other than that, LGTM. Please take a look when you have time!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you! 🎉

File: core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitAdminIntegrationTestWithJdbcDatabase.java (hunk 1 above)
Comment (laysakura) [reply core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitAdminIntegrationTestWithJdbcDatabase.java]: 68ea3f1d30d63666c92929f62cb225869372becb

File: core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitAdminIntegrationTestWithJdbcDatabase.java [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
 import com.scalar.db.transaction.consensuscommit.ConsensusCommitAdminIntegrationTestBase;
 import java.util.Properties;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIf;
 
 public class ConsensusCommitAdminIntegrationTestWithJdbcDatabase
     extends ConsensusCommitAdminIntegrationTestBase {
 
+  private static final String PROP_JDBC_URL = "scalardb.jdbc.url";
+  private static final String DEFAULT_JDBC_URL = "jdbc:mysql://localhost:3306/";
```
Comment (brfrn169) [inline core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitAdminIntegrationTestWithJdbcDatabase.java:13]: Can you please also delete these?

Comment (laysakura) [reply core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitAdminIntegrationTestWithJdbcDatabase.java]: Sorry, I did it: d790e90c

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

One thing, I just noticed this PR has `docs/getting-started/sample.db`. Do we really need this file?

**Review by komamitsu — APPROVED** [review]
LGTM, thanks!

**Comment by laysakura** [comment]
@brfrn169 
> One thing, I just noticed this PR has docs/getting-started/sample.db. Do we really need this file?

Good catch! I removed it in 423b06fc.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**